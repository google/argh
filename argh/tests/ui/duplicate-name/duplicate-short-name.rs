/// Command
#[derive(argh::FromArgs)]
struct Cmd {
    /// foo1
    #[argh(option, short = 'f')]
    foo1: u32,

    /// foo2
    #[argh(option, short = 'f')]
    foo2: u32,

    /// bar1
    #[argh(option, short = 'b')]
    bar1: u32,

    /// bar2
    #[argh(option, short = 'b')]
    bar2: u32,
}

fn main() {}
