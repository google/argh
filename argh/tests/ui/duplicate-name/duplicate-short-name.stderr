error: The short name of "-f" was already used here.
 --> tests/ui/duplicate-name/duplicate-short-name.rs:4:5
  |
4 | /     /// foo1
5 | |     #[argh(option, short = 'f')]
6 | |     foo1: u32,
  | |_____________^

error: Later usage here.
  --> tests/ui/duplicate-name/duplicate-short-name.rs:8:5
   |
8  | /     /// foo2
9  | |     #[argh(option, short = 'f')]
10 | |     foo2: u32,
   | |_____________^

error: The short name of "-b" was already used here.
  --> tests/ui/duplicate-name/duplicate-short-name.rs:12:5
   |
12 | /     /// bar1
13 | |     #[argh(option, short = 'b')]
14 | |     bar1: u32,
   | |_____________^

error: Later usage here.
  --> tests/ui/duplicate-name/duplicate-short-name.rs:16:5
   |
16 | /     /// bar2
17 | |     #[argh(option, short = 'b')]
18 | |     bar2: u32,
   | |_____________^
